graphistry/plugins/sentinel.py

```diff
@@ -0,0 +1,699 @@
+        with a warning. Set single_table=False to get all result tables.
+
+        :param query: KQL query string to execute
+        :type query: str
+        :param timespan: Time range for the query (default: 24 hours)
+        :type timespan: Optional[Union[timedelta, tuple[datetime, datetime]]]
+        :param unwrap_nested: Strategy for handling nested/dynamic columns
+        :type unwrap_nested: Optional[bool]
+        :param single_table: If True, return single DataFrame; if False, return list
+        :type single_table: bool
+        :param include_statistics: Include query statistics in DataFrame attrs
+        :type include_statistics: bool
+        :returns: Single DataFrame if single_table=True, else list of DataFrames
+        :rtype: Union[pd.DataFrame, List[pd.DataFrame]]
+
+        **unwrap_nested semantics:**
+
+        - **True**: Always attempt to unwrap nested columns; raise on failure
+        - **None**: Use heuristic - unwrap if the result looks nested
+        - **False**: Never attempt to unwrap nested columns
+
+        **Example: Basic security query (single table mode)**
+            ::
+
+                import graphistry
+                from datetime import timedelta
+                g = graphistry.configure_sentinel(...)
+
+                query = '''
+                SecurityEvent
+                | where TimeGenerated > ago(1d)
+                | where EventID == 4625  // Failed logon
+                | project TimeGenerated, Account, Computer, IpAddress
+                | take 1000
+                '''
+
+                # Query last 7 days
+                df = g.kql(query, timespan=timedelta(days=7))
+                print(f"Found {len(df)} failed logon events")
+
+        **Example: Get all tables as list**
+            ::
+
+                # Always get a list of all tables
+                dfs = g.kql(query, single_table=False)
+                df = dfs[0]
+
+        **Example: Query with specific time range**
+            ::
+
+                from datetime import datetime, timedelta
+
+                # Query specific time window
+                start = datetime(2024, 1, 1)
+                end = datetime(2024, 1, 7)
+                df = g.kql(query, timespan=(start, end))
+
+        **Example: Multi-table query**
+            ::
+
+                query = '''
+                SecurityEvent | summarize Count=count() by EventID | top 5 by Count;
+                SecurityAlert | take 10
+                '''
+
+                # With single_table=False, returns all tables
+                frames = g.kql(query, single_table=False)
+                events_df = frames[0]
+                alerts_df = frames[1]
+        """
+        results = self._sentinel_query(query, timespan=timespan)
+
+        if not results:
+            if single_table:
+                raise ValueError("Query returned no results")
+            return []
+
+        dfs: List[pd.DataFrame] = []
+
+        for result in results:
```

Comment:
can most of the wrangling in this file go away and be reused from kusto, so we're not maintaining 2 clones?

i'm not sure how close/different they are and how

ex: if just the initial connection changes, then we can parameterize kusto by the client connection